v5.2.15. Ported to python 3

juokelis opened the pull request:
Fixes #22 as requested by @zdzichu :)

juokelis committed:
v5.2.15. Ported to python 3

juokelis commented:
Not tested though. Please come back if something wrong.

juokelis committed:
Missed one place. Killed checking while not playing feature I think

smortex commented:
I needed to tune this too:

```diff
diff --git a/service.py b/service.py
index 1d9f004..ac9ea15 100644
--- a/service.py
+++ b/service.py
@@ -88,7 +88,7 @@ def Download(id,url,format,stack=False):
     zip = os.path.join( __temp__, "OpenSubtitles.zip")
     f = urllib.urlopen(url)
     if not os.path.exists( __temp__ ):
-        os.mkdir( __temp__, mode=0775 )
+        os.mkdir( __temp__, mode=0o775 )
     with open(zip, "wb") as subFile:
       subFile.write(f.read())
     subFile.close()
```

Then Kodi is able to download subtitles :tada:

smortex committed:
Fix octal notation for Python 3

Accotding to pep-3127, the gramar for an octal number is `"0" ("o" |
"O") octdigit+`

https://www.python.org/dev/peps/pep-3127/

This fixes:

```
2020-12-28 20:11:38.452 T:102161   ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'SyntaxError'>
                                                   Error Contents: invalid token (service.py, line 91)
                                                     File "/home/romain/.kodi/addons/service.subtitles.opensubtitles/service.py", line 91
                                                       os.mkdir( __temp__, mode=0775 )
                                                                                   ^
                                                   SyntaxError: invalid token
                                                   -->End of Python script error report<--

2020-12-28 20:11:38.476 T:102161    INFO <general>: Python interpreter stopped
2020-12-28 20:11:38.487 T:101946   ERROR <general>: GetDirectory - Error getting plugin://service.subtitles.opensubtitles_by_opensubtitles/?action=search&languages=English%2cFrench&preferredlanguage=Undetermined
```

juokelis committed:
Merge pull request #1 from smortex/python3-mode

Fix octal notation for Python 3

aanderse commented:
Any chance we can get a merge here?

Thanks for your work on this @juokelis!

smortex commented:
@aanderse according to https://github.com/juokelis/service.subtitles.opensubtitles_by_opensubtitles/pull/1#issuecomment-752057129, the code is now hosted here: https://github.com/juokelis/service.subtitles.opensubtitles. In order to use it with a recent version of Kodi, I had to tweak it a bit, see PR in that project if needed and maybe provide feedback to it if you give it a try.

Thanks!
